### DISCUSSION TITLE
Responding to SlashCommand and updating message contents

### DISCUSSION THREAD

**Discussion by rlightner** [stoiveyp/Slack.NetStandard#78] [General]
I'm responding with a parsed SlashCommand and sending a "looking..." message.

I then respond a few seconds later with another InteractiveMessage and set  `ReplaceOriginal = true` but the original message is not being replaced. Has anyone successfully done this before?

pseudo-code:

``` c#
var initialResponse = new InteractionMessage(ResponseType.Ephemeral)
{
    Text = "initial response..."
};

await command.Respond(initialResponse);

var finalResponse = new InteractionMessage(ResponseType.Ephemeral)
{
    ReplaceOriginal = true,
    Text = "my final response."
};

await command.Respond(finalResponse);
```

**Comment by stoiveyp** [comment]
Hi @rlightner - this is an interesting one, and I admit the Slack documentation doesn't make it very clear.

The issue here is that the `.Respond()` method you're invoking is in response to the Slash Command both times - so you are actually sending two messages to the Slash command response (perfectly valid scenario) and that's why you're seeing both messages.

To update an ephemeral message, the user has to interact with the message in some way (stated in the Slack docs in the last paragraph of this section: https://api.slack.com/messaging/managing#ephemeral ). This would generate an interaction in your app such as a BlockActionPayload - and that payload would have its own, different, ResponseUrl.

This is the responseUrl that would allow ReplaceOriginal to work the way you're hoping. But it can't be done purely from the Slash Command.

Hope this makes things a bit clearer

**Reply by rlightner** [reply]
Yup, thanks for the response!
